### PR TITLE
files cache key eviction

### PR DESCRIPTION
**Pull request by shimonewman**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/tasks/maintenance.php [hunk 1]
```diff
@@ -118,24 +118,33 @@ function renewCertificates($dbForConsole)
             }
         }
 
+        function filesCacheKeyEviction()
+        {
+            /**
+              *  Delete cache keys who did not modify for the last 90 days.
+             */
+            shell_exec('find '.APP_STORAGE_CACHE.' -type f ! -mtime -90 -print0|xargs -0 rm -f');
```
Comment (TorstenDittmann) [inline app/tasks/maintenance.php:126]: We probably need to check the performance on like 1 million files 🤔

Comment (rwese) [reply app/tasks/maintenance.php:126]: The current way would hit the max length limit for commands, causing the truncated filenames leading to weird not-found errors.

If there was no special reason to use `rm -f`, like files created with wrong permissions or user, using find's `-delete` switch should work _pretty much_ regardless of the amount of files.

Long runtimes must be considered though.

**Comment by christyjacob4** [comment]
@shimonewman  closing this now since we decided to go ahead with another approach .. Feel free to re-open if you feel necessary

**Comment by christyjacob4** [comment]
Closed in favour of #3491

**Pull request closed without merging**